src/components/Layout/GridTableLayout/GridTableLayout.stories.tsx

```diff
@@ -0,0 +1,232 @@
+import { Meta } from "@storybook/react";
+import { useEffect, useMemo, useState } from "react";
+import { checkboxFilter, multiFilter } from "src/components/Filters";
+import { GridDataRow } from "src/components/Table";
+import { collapseColumn, column, numericColumn, selectColumn } from "src/components/Table/utils/columns";
+import { simpleHeader } from "src/components/Table/utils/simpleHelpers";
+import { noop } from "src/utils";
+import { withBeamDecorator, withRouter, zeroTo } from "src/utils/sb";
+import { TestProjectLayout } from "../Layout.stories";
+import { GridTableLayout as GridTableLayoutComponent, useGridTableLayoutState } from "./GridTableLayout";
+
+export default {
+  component: GridTableLayoutComponent,
+  decorators: [withBeamDecorator, withRouter()],
+  parameters: { layout: "fullscreen" },
+} satisfies Meta;
+
+type Data = { name: string | undefined; value: number | undefined; status: string; priority: number };
+type HeaderRow = { kind: "header"; id: string; data: undefined };
+type ParentRow = { kind: "parent"; id: string; data: Data; children: GridDataRow<Row>[] };
+type DataRow = { kind: "data"; id: string; data: Data };
+type Row = HeaderRow | ParentRow | DataRow;
+
+export function GridTableLayout() {
+  const filterDefs = useMemo(() => getFilterDefs(), []);
+  const columns = useMemo(() => getColumns(), []);
+
+  const layoutState = useGridTableLayoutState({
+    persistedFilter: {
+      filterDefs,
+      storageKey: "grid-table-layout",
+    },
+    useSearch: "client",
+  });
+
+  return (
+    <TestProjectLayout>
+      <GridTableLayoutComponent
+        pageTitle="Grid Table Layout Example"
+        breadcrumb={[
+          { href: "/", label: "Home" },
+          { href: "/", label: "Sub Page" },
+        ]}
+        layoutState={layoutState}
+        tableProps={{
+          columns: [collapseColumn<Row>(), selectColumn<Row>(), ...columns],
+          rows: [simpleHeader, ...makeNestedRows(3)],
+          sorting: { on: "client", initial: [columns[1].id!, "ASC"] },
+        }}
+        primaryAction={{ label: "Primary Action", onClick: noop }}
+        secondaryAction={{ label: "Secondary Action", onClick: noop }}
+        tertiaryAction={{ label: "Tertiary Action", onClick: noop }}
+      />
+    </TestProjectLayout>
+  );
+}
+
+export function QueryTableLayout() {
+  const filterDefs = useMemo(() => getFilterDefs(), []);
+
+  const layoutState = useGridTableLayoutState({
+    persistedFilter: {
+      filterDefs,
+      storageKey: "grid-table-layout",
```

Comment:
Tangent thought: I think we've eventually like these persistedFilter/etc things to "just work" and infer their storageKey/etc based on URL path.

Reply:
yeah that is an interesting point to consider, would be cool if we can spend more focused time on our standard filter behaviors. We know we have to come back and update the UX as well since I totally punted on the new filter design to get this PR out quickly